macros/src/main/scala/MacroCompiler.scala

```diff
@@ -298,6 +298,13 @@ class MacroCompilerPass(mems: Option[Seq[Macro]],
         val name = s"mem_${i}_${j}"
         // Create the instance.
         stmts += WDefInstance(NoInfo, name, lib.src.name, lib.tpe)
+        stmts ++= lib.sortedPorts flatMap { port =>
+          port.ports flatMap (p => p.direction match {
+            case Input =>
+              Some(IsInvalid(NoInfo, WSubField(WRef(name), p.name)))
+            case Output => None
```

Comment:
This looks like a  good place for collect!
```scala
ports.collect { case Port(_, pname, Input, _) => IsInvalid(NoInfo, WSubField(WRef(name), pname)) }
```

Reply:
!